### ISSUE TITLE
get_full_cea_output returns different results from other methods 

### ISSUE THREAD

**Issue by janstruzinski** [sonofeft/RocketCEA#47]
Hi, 

I am currently writing a library for sizing of staged combuston cycles and I noticed some discrepancies in RocketCEA. 

The mole fractions from get_full_cea_output and from get_SpeciesMoleFractions are not the same. I also noticed that MW calculations vary a lot between different methods. For the code below, get_full_cea_output gives 15.987 g/mol in the chamber, while get_Chamber_MolWt_gamma gives 17.02 g/mol.  When CEA_Obj from cea_obj_w_units is used, get_Chamber_MolWt_gamma gives 17.33 g/mol (so molecular weight from get_full_cea_output as if graphite products had no volume). When I calculate it manually for molar fractions from get_SpeciesMoleFractions I get 15.45 g/mol - a fairly big difference. 

Here is the code that compares the values. 

```python

import rocketcea.cea_obj as rcea
from rocketcea.cea_obj_w_units import CEA_Obj
# Molar fractions issue
CEA_obj = rcea.CEA_Obj(oxName="LOX", fuelName="CH4")
print(CEA_obj.get_full_cea_output(Pc=800, MR=0.2, pc_units="bar", short_output=1, output="si"))
print(CEA_obj.get_SpeciesMoleFractions(Pc=800, MR=0.2))
# Molar weight issue
print(CEA_obj.get_Chamber_MolWt_gamma(Pc=800, MR=0.2))
CEA_obj = CEA_Obj(oxName="LOX", fuelName="CH4", isp_units='sec', cstar_units='m/s',
                  pressure_units='bar', temperature_units='K', sonic_velocity_units='m/s',
                  enthalpy_units='kJ/kg', density_units='kg/m^3', specific_heat_units='kJ/kg-K',
                  viscosity_units='millipoise', thermal_cond_units='W/cm-degC')
print(CEA_obj.get_Chamber_MolWt_gamma(Pc=800, MR=0.2))

``` 
Do you know what may be an issue? Thanks for the help!

With kind regards, 
Jan

**Comment by janstruzinski** [comment]
The molecular weight calculated manually from molar fractions given by get_full_cea_output agrees with the one given by it. So the issue must be with other methods I think.

**Comment by janstruzinski** [comment]
I noticed I gave Pc in bar, and not psi, to rcea.CEA_Obj. Now the methods give consistent answers when it comes to mole fractions, but MW they give (17.33 g/mol) is different from the one given by get_full_cea_output (15.987 g/mol). I think this is because they account for graphite mole fraction (17.33 = 15.987 / (1-0.077)), such that MW can be used in ideal gas law. But this value is not an actual molecular weight. Rather, it is an implementation of compressibility factor of 0 for condensibles/solids in the products, while MW should stay constant regardless of condensibles/solids fraction. I think it would be much clearer if methods returned actual MW value. In NASA RP 1311, a distinction was introduced between MW (actual molecular weight) and M (molecular weight fixed for use in ideal gas model). Maybe this could be implemented here as well?

**Comment by sonofeft** [comment]
RocketCEA is a wrapper for the NASA CEA code.
Are you saying that you disagree with the numbers that the NASA CEA code is generating or is there a problem with RocketCEA?
I know that getting units input correctly can often be a problem.

**Comment by janstruzinski** [comment]
In NASA CEA, MW is used for actual molecular weight. This is what is also given in get_full_output. M is used for MW fixed for condensibles/solids for use in ideal gas model (in other words, M is used for a product of MW and mixture compressibility). 

In RocketCEA, MW is used for the latter, and not for actual molecular weight. As the result, if you use methods like get_Chamber_MolWt_gamma, you get different  MW value than from NASA CEA. 

I think it can very confusing. Could it be fixed in the code to ensure the same naming convention as for CEA or the difference mentioned in the documentation?

**Comment by sonofeft** [comment]
Thank you... I see the problem now.
Yes... that can be fixed.
I don't think that many users have condensibles to deal with, but I still need to think about a graceful transition.
Thanks again.

**Comment by sonofeft** [comment]
Fixed in new version 1.2.1
